src/System.Threading.Tasks/tests/Task/TaskAPMTest.cs

```diff
@@ -18,101 +19,120 @@ namespace System.Threading.Tasks.Tests
     /// <summary>
     /// A class that implements the APM pattern to ensure that TPL can support APM patttern
     /// </summary>
-    public sealed class TaskAPMTest
+    public sealed class TaskAPMTests : IDisposable
     {
         /// <summary>
         /// Used to indicate whether to test TPL's Task or Future functionality for the APM pattern
         /// </summary>
-        private readonly bool _hasReturnType;
+        private bool _hasReturnType;
 
         /// <summary>
         /// Used to synchornize between Main thread and async thread, by blocking the main thread until
         /// the thread that invokes the TaskCompleted method via AsyncCallback finishes
         /// </summary>
-        private ManualResetEvent _mre;
+        private ManualResetEvent _mre = new ManualResetEvent(false);
 
-        private const int INTINPUT = 1000;     // the input to the LongTask<int>.DoTask/BeginDoTask
         /// <summary>
-        /// The constant that defines the amount time to spinwait (to simulate work) in the LongTask class
+        /// The input value to LongTask<int>.DoTask/BeginDoTask
         /// </summary>
-        private const int LongTaskSeconds = 5;
+        private const int IntInput = 1000;
 
         /// <summary>
-        /// Ctor that reads the XML testcase and sets the boolean to indicate whether to test Task or Future
-        /// functionality
+        /// The constant that defines the amount time to spinwait (to simulate work) in the LongTask class
         /// </summary>
-        /// <param name="hasReturnType"></param>
-        public TaskAPMTest(bool hasReturnType)
+        private const int LongTaskMilliseconds = 1000;
```

Comment:
I think we should decrease this value.  Each test is spinning up a Task that spins for this long.